### PR TITLE
Hardening 2

### PR DESCRIPTION
**Pull request by gregaubert**
* [x] Stop rounding numbers in graph tooltips of project activity page
* [x] [SONAR-9474](https://jira.sonarsource.com/browse/SONAR-9474)
* [x] [SONAR-8689](https://jira.sonarsource.com/browse/SONAR-8689)
* [x] [SONAR-9513](https://jira.sonarsource.com/browse/SONAR-9513)
* [x] [SONAR-9719](https://jira.sonarsource.com/browse/SONAR-9719)

### CODE REVIEW

File: server/sonar-web/src/main/js/apps/projectActivity/components/GraphHistory.js [hunk 1]
```diff
@@ -64,6 +64,8 @@ export default class GraphHistory extends React.PureComponent {
   formatValue = (tick /*: string | number */) =>
     formatMeasure(tick, getShortType(this.props.metricsType));
 
+  formatTooltipValue = (tick /*: string | number */) => formatMeasure(tick, this.props.metricsType);
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/projectActivity/components/GraphHistory.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

File: server/sonar-web/src/main/js/apps/settings/components/DefinitionDefaults.js [hunk 2]
```diff
@@ -19,21 +19,71 @@
...
-    onReset: PropTypes.func.isRequired
+  /*:: props: Props; */
+  state /*: State*/ = { reseting: false };
+
+  handleClose = () => {
+    this.setState({ reseting: false });
   };
 
-  handleReset(e /*: Object */) {
+  handleReset = (e /*: Event & {target: HTMLElement} */) => {
```
Comment (SonarTech) [inline server/sonar-web/src/main/js/apps/settings/components/DefinitionDefaults.js]: MINOR Remove parentheses around the parameter of this arrow function. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=javascript%3AS3524)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-web/src/main/js/components/issue/Issue.js [hunk 3]
```diff
@@ -116,6 +116,13 @@ export default class BaseIssue extends React.PureComponent {
       this.togglePopup('edit-tags');
       return false;
     });
+    key('space', 'issues', () => {
+      if (this.props.onCheck) {
+        this.props.onCheck(this.props.issue.key);
+        return false;
+      }
+      return true;
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/components/issue/Issue.js]: I don't think you need to return anything in this case.

File: server/sonar-web/src/main/js/helpers/handlebars/ifShowAvatars.js [hunk 4]
```diff
@@ -1,30 +0,0 @@
...
- */
-function enableGravatar() {
-  const getStore = require('../../app/utils/getStore').default;
-  const { getSettingValue } = require('../../store/rootReducer');
-
-  const store = getStore();
-  return (getSettingValue(store.getState(), 'sonar.lf.enableGravatar') || {}).value === 'true';
-}
-
-module.exports = function(options) {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/helpers/handlebars/ifShowAvatars.js]: 😌 <- my face every time we remove some backbone/handlebars code

File: server/sonar-web/src/main/js/apps/settings/components/DefinitionDefaults.js [hunk 5]
```diff
@@ -19,21 +19,71 @@
...
     e.preventDefault();
     e.target.blur();
+    this.setState({ reseting: true });
+  };
+
+  handleSubmit = () => {
```
Comment (stas-vilchik) [inline server/sonar-web/src/main/js/apps/settings/components/DefinitionDefaults.js]: 💀  you forgot to `preventDefault`

**Comment by gregaubert** [comment]
@stas-vilchik Fixed!

**Pull request merged**